database/src/desktop/view/event.h

```diff
@@ -90,16 +91,13 @@ struct Event {
   Path path;
 
   // If the Event is a cancel event, the event registration is removed from the
-  // View it is attached to. Since they are stored in UniquePtrs this would
-  // normally mean they get deallocated when they're removed, but we need it to
-  // live long enough to fire the Event. So instead, the event takes ownership
-  // of the pointer.
+  // View it is attached to. Since they are stored in std::shared_ptrs, they are
+  // only deallocated when all references are gone.
   //
-  // This field is not used to access any data, it's only used
-  // to take ownership. To keep the code streamlined, any time the pointer is
-  // needed, owned or not, it will go through the event_registration field
-  // above.
-  UniquePtr<EventRegistration> event_registration_ownership_ptr;
+  // This field is not used to access any data, it's only used to take
+  // ownership. To keep the code streamlined, any time the pointer is needed,
+  // owned or not, it will go through the event_registration field above.
+  std::shared_ptr<EventRegistration> event_registration_ownership_ptr;
```

Comment:
This has been changed from a UniquePtr to a shared_ptr, because Events are copied around all over the place in this library. Making it shared is much safer than trying to modify the entire desktop RTDB library to use move semantics for events. (And in fact there are certain places where we can't, such as a vector.insert(.....) that must be able to copy elements, and not move them.